src/gui/sidebar/previews/page/SidebarPreviewPages.cpp

```diff
@@ -187,11 +187,6 @@ void SidebarPreviewPages::updatePreviews() {
     doc->lock();
     size_t len = doc->getPageCount();
 
-    if (this->previews.size() == len) {
```

Comment:
This is the part that resolves the related issue. Comparing the sizes here gives no clue about whether the document is new. We would need to provide that as argument to the listener, hence I removed the whole block.